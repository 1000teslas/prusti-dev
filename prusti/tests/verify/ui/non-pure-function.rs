#![allow(dead_code)]
extern crate prusti_contracts;

mod foo {
    pub fn get_false() -> bool {
        false
    }
}

fn get_true() -> bool {
    true
}

#[requires="get_true() && !foo::get_false()"]
fn main() {
}
