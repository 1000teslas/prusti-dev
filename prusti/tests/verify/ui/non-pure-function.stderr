error: [Prusti encoding] use of impure function "get_true" in assertion
  --> $DIR/non-pure-function.rs:14:13
   |
14 | #[requires="get_true() && !foo::get_false()"]
   |             ^^^^^^^^^^

error: [Prusti encoding] use of impure function "foo::get_false" in assertion
  --> $DIR/non-pure-function.rs:14:28
   |
14 | #[requires="get_true() && !foo::get_false()"]
   |                            ^^^^^^^^^^^^^^^^

error: [Prusti] use of impure function might be reachable.
  --> $DIR/non-pure-function.rs:14:13
   |
14 | #[requires="get_true() && !foo::get_false()"]
   |             ^^^^^^^^^^
   |
   = help: Functions called from assertions should be marked as pure.

error: aborting due to 3 previous errors

